views.py

```diff
@@ -450,6 +457,118 @@ def eventCheckInOut(request, event_id):
         return render(request, 'coderdojomobile/eventCheckInOut.html', context)
 
 
+def generate_uuid():
+    now = datetime.datetime.now()
+    rand = random.randint(1, 999)
+    micro = now.strftime("%f").zfill(6)[3:6]
+    uuid = now.strftime("%M%S") + str(micro) + str(rand).zfill(3)
+    return uuid
+
+
+def import_ticket(ticket, event):
+    '''
+    if badge id is not null, it is used to identify the participant
+    otherwise, name and surname are used
+    if the badge id does not match an existing badge id, a new participant
+    is created. This is done to allow Ninjas to use the same badge in more
+    than one Dojo.
+    No check is performed on name - surname w.r.t badge id.
+    '''
+    name = ticket['Name']
+    surname = ticket['Surname']
+    ticket_id = ticket['Ticket_id']
+    badge_id = ticket['Badge_id']
+    participant = None
+    if badge_id is not None and len(badge_id) > 0:
+        # Look for the Participant
+        try:
+            participant = Participant.objects.get(uuid=badge_id)
+        except Participant.DoesNotExist:
+            None
+    if participant is None:
+        # create
+        badge_id = generate_uuid()
+        participant = Participant.objects.create(
+                                                name=name,
+                                                surname=surname,
+                                                uuid=badge_id
+                                                )
+    if ticket_id is None or len(ticket_id) == 0:
+        ticket_id = generate_uuid()
+    ticket = Ticket.objects.create(
+                                  participant=participant,
+                                  event=event,
+                                  has_checked_in=False,
+                                  uuid=ticket_id
+                                  )
+    return ticket
+
+
+def import_event_participants(external_ticket_list, event):
+    processed_tickets = []
+    for ticket in external_ticket_list:
+        processed_tickets.append(import_ticket(ticket, event))
+    return processed_tickets
+
+
+@permission_required(
+    'coderdojomobile.change_ticket',
+    login_url=reverse_lazy('coderdojomobile:login',
+                           current_app="coderdojomobile")
+)
+def event_ticket_upload(request, event_id):
+    context = base_function(request)
+    event = None
+    processed_tickets = []
+    if request.method == 'POST':  # TODO handle get
+        form = TicketUploadForm(request.POST, request.FILES)
+        '''
+        CSV format support 4 fields:
+        name, surname, ticket id (optional), badge id (optional)
+        '''
+        ticket_list = []
+        if form.is_valid():
+            try:
+                event = Event.objects.get(id=event_id)
+            except Event.DoesNotExist:
+                None
```

Comment:
Please use "pass" if nothing to be done. I would have preferred to see some error on the screen.